x/cosmwasmpool/client/cli/query.go

```diff
@@ -0,0 +1,42 @@
+package cli
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/osmosis-labs/osmosis/osmoutils/osmocli"
+	"github.com/osmosis-labs/osmosis/v16/x/cosmwasmpool/client/queryproto"
+	"github.com/osmosis-labs/osmosis/v16/x/cosmwasmpool/types"
+)
+
+// NewQueryCmd returns the cli query commands for this module.
+func NewQueryCmd() *cobra.Command {
+	cmd := osmocli.QueryIndexCmd(types.ModuleName)
+
+	qcGetter := queryproto.NewQueryClient
+	osmocli.AddQueryCmd(cmd, qcGetter, GetCmdPools)
+	osmocli.AddQueryCmd(cmd, qcGetter, GetCmdContractInfoByPoolId)
+	cmd.AddCommand(
+		osmocli.GetParams[*queryproto.ParamsRequest](
+			types.ModuleName, queryproto.NewQueryClient),
+	)
+
+	return cmd
+}
+
+func GetCmdPools() (*osmocli.QueryDescriptor, *queryproto.PoolsRequest) {
+	return &osmocli.QueryDescriptor{
```

Comment:
Looking into this for the serialization issue we discussed offline. IIUC, I think the best solution is to modify osmocli to allow us to manipulate the response before returning it (that way we can cast it to the proto version instead of the struct version)